### ISSUE TITLE
Understanding Object Destructuring

### ISSUE THREAD

**Issue by dylanmorris51** [nss-day-cohort-46/How-to-Ask-for-Help#67]
**_Ticket MUST be completed before getting help from an instructor_**

#### Have you searched the other issue tickets? If not, do that first. 
- [x ] Yes
- If No, stop and search.

####  REQUIRED Add Tags to Issue Ticket and Assign to Project Board
 

- [x] Added label to specific project/assignment that you need help with
- [x] Added to `c46 Help-Thread` project board

#### REQUIRED Describe your issue:
>> I want to understand object destructuring; specifically why we declare variables within curly braces {} when 
we useContext in the List components 

```
 const { animals, getAnimals } = useContext(AnimalContext)
 ```
 
#### REQUIRED What have you googled? **_(You must have at least 3 links related to your issue)_**

- [x] 1.  https://www.freecodecamp.org/news/array-and-object-destructuring-in-javascript/
- [ ] 2. https://daveceddia.com/usecontext-hook/
- [ ] 3. https://www.freecodecamp.org/news/how-to-destructure-the-fundamentals-of-react-hooks-d13ff6ea6871/

#### REQUIRED: What have you tried? Tell us everything you have tried. 
>> I tried reading these articles

#### REQUIRED: Copy/paste link TO THE FILE on your branch that you are having issues with

**Your link:** 

https://github.com/dylanmorris51/kennels/blob/a259f214b07a59661ac30755d05a82aa75d50bad/src/components/animal/AnimalList.js#L10

#### Code Snippet

```
const { animals, getAnimals } = useContext(AnimalContext)
```

**Comment by Scott47** [comment]
@dylanmorris51 That third article is probably the best you can get on destructing react hooks. It does neglect the details of React context API because it is a little different and, like the article says, 'out of scope'.  As I understand from the docs, you destructure in the component that consumes the context to make each context a separate node in the tree and that keeps the re-rendering process efficient. This is about the extent of my understanding re: react context api

**Comment by Younique98** [comment]
I had this same issue. I had forgotten to wrap my EmployeeList with the LocationProvider within the ApplicationView. I kept getting a deconstructing error message in the browser.
